helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -86,11 +87,17 @@
   private ClusterConfig _clusterConfig;
 
   private boolean _updateInstanceOfflineTime = true;
+  private PauseSignal _pauseSignal;
   private MaintenanceSignal _maintenanceSignal;
-  private boolean _isMaintenanceModeEnabled;
+  boolean _isControllerPaused;
```

Comment:
Question, why we need these 2 boolean values while we already have the signals in the cache?
Won't it be easy to derive these 2 statuses by checking the signals?